src/main/java/org/victorrobotics/frc/dtlib/actuator/motor/DTMotor.java

```diff
@@ -0,0 +1,135 @@
+package org.victorrobotics.frc.dtlib.actuator.motor;
+
+import org.victorrobotics.frc.dtlib.function.supplier.DTLimitedBooleanSupplier;
+import org.victorrobotics.frc.dtlib.function.supplier.DTLimitedDoubleSupplier;
+import org.victorrobotics.frc.dtlib.function.supplier.DTLimitedLongSupplier;
+import org.victorrobotics.frc.dtlib.function.supplier.DTLimitedSupplier;
+import org.victorrobotics.frc.dtlib.network.DTSendable;
+
+import edu.wpi.first.util.sendable.SendableBuilder;
+
+public interface DTMotor<MOTORTYPE, CURRENTLIMITTYPE> extends DTSendable, AutoCloseable {
```

Comment:
The idea is that clients can declare motors simply as a DTMotor, not parameterized, or as DTMotor<?, ?>, unless they need the relevant advanced functionality. I could try to make a common interface for current limits, but this is not possible for motors.

Reply:
It would be great if we can create a client that uses the `DTMotor` class and be sure that it's simple to use. Most of the programmers on our team have limited Java knowledge and we should not expose them to generics, even if they are used in the implementation of the library. Two possibilities come to mind:
* DTMotor is a non-generic interface. Client code is written to use `DTMotor`, except where they create the motor object (`DTMotor motor - new DTTalonFX(...)`).
* DTMotor is a generic interface. Clients use `DTTalonFX motor = new DTTalonFX(...)`.

Reply:
Currently, both of these instantiations are perfectly valid. Even though DTMotor is a generic interface, it is not required to be parameterized; that simply means that its parameter types default to `Object`. For the internal motor objects, I think it’s probably better to parameterize them, and students won’t need to call `internal()` unless they need the advanced functionality provided by the underlying class.

Reply:
The true motivation behind using an interface is so swapping out motors is basically as easy as changing the constructor used, and maybe the declared type (if not left as `DTMotor`). All of the methods are identical in name and signature, so it’s easy to transfer your abilities from a `DTTalonFX` directly to a `DTNeo`.